client/app/bundles/course/assessment/question/programming/actions/onlineEditorActionCreators.jsx

```diff
@@ -33,25 +33,81 @@ export function deletePythonTestCase(testType, index) {
   };
 }
 
-export function updateNewDataFile(filename, index) {
+export function updatePythonNewDataFile(filename, index) {
```

Comment:
For the new data files, is there a need to define a new action name for the different programming languages? 

My opinion is if updating new data files for both Python and C++ are the same, no need to explicitly define another function and name. This is unless you can think of possible use cases for other languages (eg. java) that require special logic for new data files. 

Otherwise this can actually be a common action for all programming languages to use.